📖 update releasing.md with more steps (#236)
metal3-io/ip-address-manager: docs/releasing.md

- For any subdirectory with `go.mod` in it (excluding `hack/tools`), create
  another Git tag with directory prefix, ie.
  `git tag -s api/v1.x.y -m api/v1.x.y`.

>   NOTE: Do not create annotated tags for go modules.

Reviewer: ```suggestion
  **NOTE:** Do not create annotated tags for go modules.
```

- If it is a release candidate (RC) or a pre-release, tick pre-release box.
- Publish the release.

## Post-release actions for new release branches

Patch release process is complete, but further additional actions are required
in the Metal3 project to take in the new major or minor release. These steps are
documented in
[Metal3 release process](https://github.com/metal3-io/metal3-docs/blob/main/processes/releasing.md)

### Branch protection rules

Branch protection rules need to be applied to the new release branch. Copy the
settings after the previous release branch, with the exception of
`Required tests` selection. Required tests can only be selected after new
keywords are implemented in Jenkins JJB, and project-infra, and have been run

> at least one time.

Reviewer: ```suggestion
at least once in the PR targeting the branch in question.
```

Author: Fixed

### Build badges

Update the build badges in the `README.md` to point to correct Jenkins jobs,
so the status of the release branch is visible.

[Example](https://github.com/metal3-io/ip-address-manager/pull/232)

## Additional actions outside this repository

Further additional actions are required in the Metal3 project after IPAM

> release. Continue with the
> [Metal3 release process](https://github.com/metal3-io/metal3-docs/blob/main/processes/releasing.md)

Reviewer: ```suggestion
release. For that, please continue following the instructions provided in
[Metal3 release process](https://github.com/metal3-io/metal3-docs/blob/main/processes/releasing.md).
```

Author: Fixed
